index.dev.js

```diff
@@ -25,56 +26,59 @@ function runInsect(fmt, line) {
   return res;
 }
 
-// Handle command line arguments
-if (process.argv.length >= 4) {
-  usage();
-} else if (process.argv.length == 3) {
-  var arg = process.argv[2];
-  if (arg === "-h" || arg === "--help") {
+// Top-level await is not supported in Node 12 and earlier.
+(async function() {
+  if (process.env.INSECT_NO_RC !== "true") {
+    function processRcLine(line) {
+      var res = runInsect(Insect.fmtPlain, line);
+      // We really only care when it breaks
+      if (res && res.msgType === "error") {
+        console.error(res.msg);
+        process.exit(1);
+      }
+    }
+
+    var [util, lineReader] = await Promise.all([import("util"), import("line-reader")]);
+    var eachLine = util.promisify(lineReader.eachLine);
+
+    try {
+      await eachLine(path.join(xdgBasedir.config, "insect/insectrc"), processRcLine);
+    } catch (err) {
+      if (err.code === "ENOENT") {
+        try {
+          var os = await import("os");
+          await eachLine(path.join(os.homedir(), ".insectrc"), processRcLine);
+        } catch (err) {
+          if (err.code !== "ENOENT") {
+            throw err;
+          }
+        }
+      } else {
+        throw err;
+      }
+    }
+  }
+
+  // Handle command line arguments
+  var args = process.argv.slice(2);
+  if (args[0] === "-h" || args[0] === "--help") {
     usage();
-  } else {
+  } else if (args.length !== 0) {
     // Execute a single command
-    var res = runInsect(Insect.fmtPlain, arg);
+    var res = runInsect(Insect.fmtPlain, args.join(" "));
     if (res.msgType === "value" || res.msgType === "info") {
       console.log(res.msg);
     } else if (res.msgType === "error") {
       console.error(res.msg);
+      process.exit(1);
```

Comment:
Should we also exit with exit code 1 when an expression passed via stdin fails? e.g. `echo asdf | insect`

Reply:
Yes, that would be great. Someone could use this as `insect < my_long_script.ins` and it would be good to signal any parser/runtime errors via the exit code

Reply:
Turns out that's already the case, so never mind. :D